lib/sidekiq/logger.rb

```diff
@@ -17,8 +17,68 @@ def ctx
     end
   end
 
+  module LoggerThreadSafeLevel
+    Logger::Severity.constants.each do |severity|
+      define_method("#{severity.downcase}?") do
+        Logger.const_get(severity) >= level
+      end
+    end
+
+    def local_level
+      Thread.current[:sidekiq_log_level]
+    end
+
+    def local_level=(level)
+      case level
+      when Integer
+        Thread.current[:sidekiq_log_level] = level
+      when Symbol, String
+        Thread.current[:sidekiq_log_level] = Logger::Severity.const_get(level.to_s.upcase)
```

Comment:
I would setup a Hash with the N levels and the default proc would log a warning and return nil, rather than looking up constants.